### ISSUE TITLE
[docs] Official apex -> torch.cuda.amp migration guide

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#52279]
I propose to add such a guide directly to https://pytorch.org/docs/stable/amp.html or at https://pytorch.org/docs/stable/notes/amp_examples.html

Common questions include - would torch.cuda.amp achieve similar memory reduction? speed? is apex O2-mode at all beneficial compared to amp?

It is often aked, e.g. https://discuss.pytorch.org/t/torch-cuda-amp-vs-nvidia-apex/74994/2 https://github.com/NVIDIA/apex/issues/818. But the amp is getting improved, so it would be good to have a current state of affairs somewhat officially published

**Comment by vadimkantorov** [comment]
Some useful examples could be complete examples of PyTorch benchmarks (so that DataParallel / DDP, gradient accumulation and such could be seen in aciton)

**Comment by vadimkantorov** [comment]
One other non-trivial question is interaction of gradient accumulation and amp: are there anything to be handled at loss scaling? So that one loss scales of different (accumulated) backward passes are the same

**Comment by vadimkantorov** [comment]
Related: https://discuss.pytorch.org/t/how-can-i-use-o2-optimization-with-torch-cuda-amp-like-apex/156449
